robot_calibration/src/sensor_data_generator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "ros/ros.h"
+#include "sensor_msgs/JointState.h"
+#include <cmath>
+#include "yaml-cpp/yaml.h"
+#include <fstream>
+
+class Listener
+{  
+  public:
+    void calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg);
+    std::string timestamp = std::to_string(ros::Time::now().toSec());
+};
+
+
+void Listener::calculateAndWriteData(const sensor_msgs::JointState::ConstPtr& msg)
+{
+  double angles[2];
+  angles[0] = msg->position[0];
+  angles[1] = msg->position[1];
+
+  double link_1_length, link_2_length;
+  link_1_length = 1.8;
+  link_2_length = 1.0;
+
+  double ee_position[] = {(link_1_length * cos(angles[0])) + (link_2_length * cos(angles[0] + angles[1])) , (link_1_length * sin(angles[0])) + (link_2_length * sin(angles[0] + angles[1]))}; 
+  
+  angles[0] += 0.044; //theta1 offset
+  angles[1] += 0.013; //theta2 offset
+
+  YAML::Emitter yaml_out_stream;
+  std::ofstream outfile;
+
+  yaml_out_stream << YAML::BeginMap;
```

Comment:
The yaml output format is confusing, and hard to read. We can discuss what this should look like in person.